ui/src/plugins/org.chromium.LoadLine2/index.ts

```diff
@@ -0,0 +1,206 @@
+// Copyright (C) 2025 The Android Open Source Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Trace} from '../../public/trace';
+import {PerfettoPlugin} from '../../public/plugin';
+import {TrackNode} from '../../public/workspace';
+import {SliceTrack} from '../../components/tracks/slice_track';
+import {SourceDataset} from '../../trace_processor/dataset';
+import {LONG, NUM, STR} from '../../trace_processor/query_result';
+
+const METRIC_QUERY = `
+INCLUDE PERFETTO MODULE chrome.loadline_2;
+
+DROP TABLE IF EXISTS _chrome_loadline2_plugin_metric;
+CREATE PERFETTO TABLE _chrome_loadline2_plugin_metric AS
+WITH all_generated_slices AS (
+  SELECT
+    page,
+    'visual' AS name,
+    story_start AS ts,
+    visual_presentation - story_start AS dur
+  FROM chrome_loadline2_stages
+
+  UNION ALL
+  SELECT
+    page,
+    'interactive' AS name,
+    story_start AS ts,
+    interactive_presentation - story_start AS dur
+  FROM chrome_loadline2_stages
+
+  UNION ALL
+  SELECT
+    page,
+    page AS name,
+    story_start AS ts,
+    story_finish - story_start AS dur
```

Comment:
(I feel like I should have asked this before, sorry) What value does story_finish bring? I feel like this could cause confusion - some people could think the final score comes from this third slice. Especially because this is in the "MetricTrack"

I'm not even clear on the semantics of this timestamp, given it is emitted after a 100ms wait
https://source.chromium.org/chromium/chromium/src/+/main:third_party/crossbench/config/benchmark/loadline2/page_config_phone.hjson;l=45;drc=47487469f296a56dbbf83007ecf0fa8e3cb04cd4